src/kademlia/traversal_algorithm.cpp

```diff
@@ -593,7 +593,7 @@ void traversal_algorithm::status(dht_lookup& l)
 	l.last_sent = last_sent;
 }
 
-void traversal_algorithm::look_for_nodes(bdecode_node const& r,bool const is_traverse)
+void traversal_algorithm::look_for_nodes(bdecode_node const& r, std::function<void(const node_endpoint&)> f)
```

Comment:
does this function need to be a member of ``traversal_algorithm``? If not, I think it should be made a free function